demos/noise_suppression_demo/python/noise_suppression_demo.py

```diff
@@ -73,27 +74,52 @@ def main():
 
     log.info('OpenVINO Inference Engine')
     log.info('\tbuild: {}'.format(get_version()))
-    ie = IECore()
+    core = Core()
 
     # read IR
     log.info("Reading model {}".format(args.model))
-    model_xml = args.model
-    model_bin = model_xml.with_suffix(".bin")
-    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+    ov_encoder = core.read_model(args.model)
 
     # check input and output names
-    input_shapes = {k: v.input_data.shape for k, v in ie_encoder.input_info.items()}
-    input_names = list(ie_encoder.input_info.keys())
-    output_names = list(ie_encoder.outputs.keys())
-
-    assert "input" in input_names, "'input' is not presented in model"
+    if len(ov_encoder.inputs) == len(ov_encoder.outputs):
+        input_shapes = {}
+        output_names = []
+
+        try:
+            data_input = ov_encoder.input('input')
+            input_shapes['input'] = data_input.shape
+        except RuntimeError:
+            raise RuntimeError('Input tensor with name \'input\' is not presented in the model')
+
+        try:
+            if ov_encoder.output('output'):
+                output_names.append("output")
+        except RuntimeError:
+            raise RuntimeError('Output tensor with name \'output\' is not presented in the model')
+
+        for const_obj in ov_encoder.inputs:
+            for name in const_obj.get_names():
+                if ("inp" in name) and ("state" in name):
+                    input_shapes[name] = const_obj.shape
+
+        for const_obj in ov_encoder.outputs:
+            for name in const_obj.get_names():
+                if ("out" in name) and ("state" in name):
+                    output_names.append(name)
```

Comment:
```suggestion
        for const_obj in ov_encoder.outputs:
            if "state" not in const_obj.get_names() and "output" not in const_obj.get_names():
                raise RuntimeError("The model expected output tensor with name 'output' or 'out_state_*'")
```

Reply:
It does not work correctly

Reply:
Yeah, there is should be also the iterator through all names from `get_names()`. The idea is still the same: we need to choose inp_states to fill `input_shapes`, verify that input tensor name is `input` or `inp_state_*` and make the corresponding check for output tensor names and raise errors if checks are not passed.  
Please, try to implement it, you might modify your version or add some changes to my code suggestions

Reply:
I have tried to combine these checks for tensors.
Please, check the last update.